### PR TITLE
UAttributes Validator Feature File

### PR DESCRIPTION
**Pull request by matthewd0123**
* .feature file for testing UAttributes Validator methods
* implementation of tests in java and python

### CODE REVIEW

**Review by neelam-kushwah — COMMENTED** [review]
https://github.com/eclipse-uprotocol/up-tck/pull/31#issuecomment-2085592137

**Comment by matthewd0123** [comment]
> I don't see any provision for providing an id or reqid in the feature file. In Test Agent, we're using Factory methods to create id/reqid, causing the validation method to always return success. How do we plan to verify negative cases for the validate id or reqid method?

In the test agent, if "id_type" or "reqid_type" is set, then the UUID will be set to the factory-generated UUID. Otherwise, the ID/ReqID won't be set:

`    if data.get("attributes"):
        attributes = dict_to_proto(data["attributes"], UAttributes())
        if attributes.sink.authority.name == "default":
            attributes.sink.CopyFrom(UUri())
    else:
        attributes = UAttributes()

    if data.get("id") == "uprotocol":
        attributes.id.CopyFrom(Factories.UPROTOCOL.create())
    elif data.get("id") == "uuid":
        attributes.id.CopyFrom(Factories.UUIDV6.create())

    if data.get("reqid") == "uprotocol":
        attributes.reqid.CopyFrom(Factories.UPROTOCOL.create())
    elif data.get("reqid") == "uuid":
        attributes.reqid.CopyFrom(Factories.UUIDV6.create())`

**Comment by neelam-kushwah** [comment]
> > I don't see any provision for providing an id or reqid in the feature file. In Test Agent, we're using Factory methods to create id/reqid, causing the validation method to always return success. How do we plan to verify negative cases for the validate id or reqid method?
> 
> In the test agent, if "id_type" or "reqid_type" is set, then the UUID will be set to the factory-generated UUID. Otherwise, the ID/ReqID won't be set:
> 
> ```
>     if data.get("attributes"):
>         attributes = dict_to_proto(data["attributes"], UAttributes())
>         if attributes.sink.authority.name == "default":
>             attributes.sink.CopyFrom(UUri())
>     else:
>         attributes = UAttributes()
> 
>     if data.get("id") == "uprotocol":
>         attributes.id.CopyFrom(Factories.UPROTOCOL.create())
>     elif data.get("id") == "uuid":
>         attributes.id.CopyFrom(Factories.UUIDV6.create())
> 
>     if data.get("reqid") == "uprotocol":
>         attributes.reqid.CopyFrom(Factories.UPROTOCOL.create())
>     elif data.get("reqid") == "uuid":
>         attributes.reqid.CopyFrom(Factories.UUIDV6.create())
> ```

If the UUID is set to the factory-generated UUID, then the validateId method will always return success. How do we plan to test cases with invalid UUIDs?

File: test_agent/python/testagent.py [hunk 1]
```diff
@@ -344,6 +351,93 @@ def handle_uuid_validate_command(json_msg):
     )
 
 
+def handle_uattributes_validate_command(json_msg: Dict[str, Any]):
```
Comment (agosh01) [inline test_agent/python/testagent.py:354]: Can this function be implemented in tck step file. Test agent are just forwarding json msg all the validation happens at test manager.

Comment (agosh01) [reply test_agent/python/testagent.py:354]: This may not be possible if we need to test specific to each language. In that case this comment is invalid

**Review by neelam-kushwah — APPROVED** [review]
LGTM

**Pull request merged**